src/lib.rs

```diff
@@ -1,22 +1,108 @@
-use csv::ReaderBuilder;
+use csv::{ReaderBuilder, Trim};
+use fluvio_smartmodule::{
+    dataplane::smartmodule::SmartModuleExtraParams, smartmodule, Record, RecordData, Result,
+};
+use heck::{ToLowerCamelCase, ToSnakeCase};
+use serde_json::{json, Value};
+use std::sync::OnceLock;
 
-use fluvio_smartmodule::{smartmodule, Result as FluvioResult, Record, RecordData};
-use std::collections::HashMap;
-use serde_json::json;
+static PARAMS: OnceLock<Params> = OnceLock::new();
+const DELIMITER_PARAM_NAME: &str = "delimiter";
+const HEADER_CASE_PARAM_NAME: &str = "header_case";
+const DEFAULT_DELIMITER: u8 = b',';
 
 #[smartmodule(map)]
-pub fn map(record: &Record) -> FluvioResult<(Option<RecordData>, RecordData)> {
+pub fn map(record: &Record) -> Result<(Option<RecordData>, RecordData)> {
```

Comment:
Make `Record` a `SmartModuleRecord` to remove a build warning.  SmartModuleRecord has a updated api vs Record for these modules.

Reply:
Done!